builtin/stash--helper.c

```diff
@@ -689,6 +695,50 @@ static int show_stash(int argc, const char **argv, const char *prefix)
 	return ret;
 }
 
+static int do_store_stash(const char *stash_msg, const char *w_commit)
+{
+	struct object_id obj;
+	int ret = 0;
+
+	ret = get_oid(w_commit, &obj);
+	if (!ret) {
+		ret = update_ref(stash_msg, ref_stash, &obj, NULL,
+			REF_FORCE_CREATE_REFLOG, 0);
+	}
+
+	if (ret && !quiet)
+		fprintf_ln(stderr, _("Cannot update %s with %s"), ref_stash, w_commit);
+
+	return ret;
+}
+
+static int store_stash(int argc, const char **argv, const char *prefix)
+{
+	const char *stash_msg = NULL;
+	const char *w_commit = NULL;
+	struct option options[] = {
+		OPT__QUIET(&quiet, N_("be quiet, only report errors")),
+		OPT_STRING('m', "message", &stash_msg, "message", N_("stash message")),
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options,
+		git_stash_helper_store_usage, PARSE_OPT_KEEP_UNKNOWN);
```

Comment:
Can you verify that this parsing sees the first option? I ask because I *think* that we need to add the `PARSE_OPT_KEEP_ARGV0` flag [here](https://github.com/ungps/git/blob/50936ea98176b64c674b1c02f87bcd875f0b3c54/builtin/stash--helper.c#L753-L754) to make subsequent `parse_options()` *not* overlook the original `argv[1]`.

Reply:
Theoretically, after the first `parse_options()` inside `cmd_stash__helper()`, the first argument (argv[0]) will be the given subcommand (i.e `store`). After the second `parse_options()` inside `store_stash()`, the first argument (argv[0]) will be the first argument after `store` (i.e for `git stash store foobar`, after the second parse, argv[0] will be `foobar`).

Now, I tested `git stash store foobar` with two printf debug messages before and after `parse_options(...)` call in `store_stash()`. Before it, argv[0] is `store`, and after it is `foobar`.

I guess it is ok. Isn't it?